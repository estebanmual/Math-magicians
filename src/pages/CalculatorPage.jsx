import React from 'react';
import Calculator from '../components/Calculator';
import style from '../styling/CalculatorPage.module.css';

function CalculatorPage() {
  return (
    <div className={style.CalculatorPage}>
      <h2 className={style.CalculatorPage__title}>Let&apos;s do some math!</h2>
      <div className={style.CalculatorPage__calculator}>
        <Calculator />
      </div>
    </div>
  );
}

export default CalculatorPage;
