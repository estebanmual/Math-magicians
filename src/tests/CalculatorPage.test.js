import React from 'react';
import { render, screen } from "@testing-library/react";
import CalculatorPage from '../pages/CalculatorPage.jsx';

it('renders correctly', () => {
  render(<CalculatorPage />);
  expect(screen.getByRole('table')).toBeInTheDocument();
});
