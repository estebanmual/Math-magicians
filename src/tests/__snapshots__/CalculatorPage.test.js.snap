// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="CalculatorPage"
>
  <h2
    className="CalculatorPage__title"
  >
    Let's do some math!
  </h2>
  <div
    className="CalculatorPage__calculator"
  >
    <table
      className="calculator"
    >
      <tbody>
        <tr>
          <td
            className="calculator__result"
            colSpan="4"
            data-testid="display"
          >
            0
             
             
          </td>
        </tr>
        <tr>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              AC
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              +/-
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              %
            </button>
          </td>
          <td
            className="calculator__button__orange"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              รท
            </button>
          </td>
        </tr>
        <tr>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              7
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              8
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              9
            </button>
          </td>
          <td
            className="calculator__button__orange"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              x
            </button>
          </td>
        </tr>
        <tr>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              4
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              5
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              6
            </button>
          </td>
          <td
            className="calculator__button__orange"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              -
            </button>
          </td>
        </tr>
        <tr>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              1
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              2
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              3
            </button>
          </td>
          <td
            className="calculator__button__orange"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              +
            </button>
          </td>
        </tr>
        <tr>
          <td
            className="calculator__button__grey"
            colSpan="2"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              0
            </button>
          </td>
          <td
            className="calculator__button__grey"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              .
            </button>
          </td>
          <td
            className="calculator__button__orange"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              =
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
